examples/native_texture.rs

```diff
@@ -1,111 +1,160 @@
-    let (window, display) = create_display(&event_loop);
-
-    let mut egui_glium =
-        egui_glium::EguiGlium::new(ViewportId::ROOT, &display, &window, &event_loop);
-
-    let png_data = include_bytes!("rust-logo-256x256.png");
-    let image = load_glium_image(png_data);
-    let image_size = egui::vec2(image.width as f32, image.height as f32);
-    // Load to gpu memory
-    let glium_texture = glium::texture::SrgbTexture2d::new(&display, image).unwrap();
-    // Allow us to share the texture with egui:
-    let glium_texture = std::rc::Rc::new(glium_texture);
-    // Allocate egui's texture id for GL texture
-    let texture_id = egui_glium
-        .painter
-        .register_native_texture(glium_texture, Default::default());
+    let event_loop = EventLoop::<()>::with_user_event().build().unwrap();
+
     // Setup button image size for reasonable image size for button container.
     let button_image_size = egui::vec2(32_f32, 32_f32);
 
-    let result = event_loop.run(move |event, target| {
+    let mut app = App {
+        graphics_context: None,
+        button_image_size,
+    };
+
+    let result = event_loop.run_app(&mut app);
+    result.unwrap()
+}
+
+struct App {
+    graphics_context: Option<GraphicsContext>,
+    button_image_size: Vec2,
+}
+
+struct GraphicsContext {
+    egui_glium_instance: egui_glium::EguiGlium,
+    _glium_texture: Rc<SrgbTexture2d>,
+    texture_id: TextureId,
+    image_size: Vec2,
+    window: winit::window::Window,
+    display: glium::Display<WindowSurface>,
+}
+
+impl ApplicationHandler for App {
+    fn resumed(&mut self, event_loop: &ActiveEventLoop) {
+        let (window, glium_display) = SimpleWindowBuilder::new()
+            .set_window_builder(Window::default_attributes().with_resizable(true))
+            .with_inner_size(800, 600)
+            .with_title("egui_glium example")
+            .build(event_loop);
+
+        let mut egui_glium_instance =
+            egui_glium::EguiGlium::new(ViewportId::ROOT, &glium_display, &window, event_loop);
+
+        let png_data = include_bytes!("rust-logo-256x256.png");
+        let image = load_glium_image(png_data);
+        let image_size = egui::vec2(image.width as f32, image.height as f32);
+        // Load to gpu memory
+        let glium_texture = glium::texture::SrgbTexture2d::new(&glium_display, image).unwrap();
+        // Allow us to share the texture with egui:
+        let glium_texture = std::rc::Rc::new(glium_texture);
+        // Allocate egui's texture id for GL texture
+        let texture_id = egui_glium_instance
+            .painter
+            .register_native_texture(Rc::clone(&glium_texture), Default::default());
+
+        self.graphics_context = Some(GraphicsContext {
+            egui_glium_instance,
+            _glium_texture: glium_texture,
+            texture_id,
+            image_size,
+            window,
+            display: glium_display,
+        });
+    }
```

Comment:
So on resumption we must recreate the GL state? Does this mean previous versions of the `egui_glium` examples could break in this regard?

Reply:
I originally thought so because [`EguiGlium::new`](https://docs.rs/egui_glium/0.26.3/egui_glium/struct.EguiGlium.html#method.new) took an `EventLoopWindowTarget` previously, which has been renamed in `winit` to `ActiveEventLoop`.

But with your suggestion to take a `&dyn HasDisplayHandle`, I think I can then pass an `EventLoop` instead. In that case, we can revert back to instantiating all this stuff at the beginning.

My understanding was winit went this direction because platforms like iOS and Android can destroy and recreate your app many times over its lifetime, which means creating and releasing graphics resources like GL context and windows and such. I might be mistaken on that point though.